kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/service/impl/ProxyAuthServiceImpl.java

```diff
@@ -0,0 +1,648 @@
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.Claim;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.kernel.auth.config.MosipEnvironment;
+import io.mosip.kernel.auth.constant.AuthConstant;
+import io.mosip.kernel.auth.constant.AuthErrorCode;
+import io.mosip.kernel.auth.constant.KeycloakConstants;
+import io.mosip.kernel.auth.dto.AccessTokenResponse;
+import io.mosip.kernel.auth.dto.AccessTokenResponseDTO;
+import io.mosip.kernel.auth.dto.AuthNResponse;
+import io.mosip.kernel.auth.dto.AuthNResponseDto;
+import io.mosip.kernel.auth.dto.AuthResponseDto;
+import io.mosip.kernel.auth.dto.AuthToken;
+import io.mosip.kernel.auth.dto.AuthZResponseDto;
+import io.mosip.kernel.auth.dto.ClientSecret;
+import io.mosip.kernel.auth.dto.KeycloakErrorResponseDto;
+import io.mosip.kernel.auth.dto.LoginUser;
+import io.mosip.kernel.auth.dto.MosipUserDto;
+import io.mosip.kernel.auth.dto.MosipUserListDto;
+import io.mosip.kernel.auth.dto.MosipUserSaltListDto;
+import io.mosip.kernel.auth.dto.MosipUserTokenDto;
+import io.mosip.kernel.auth.dto.PasswordDto;
+import io.mosip.kernel.auth.dto.RIdDto;
+import io.mosip.kernel.auth.dto.RealmAccessDto;
+import io.mosip.kernel.auth.dto.RolesListDto;
+import io.mosip.kernel.auth.dto.TimeToken;
+import io.mosip.kernel.auth.dto.UserDetailsResponseDto;
+import io.mosip.kernel.auth.dto.UserNameDto;
+import io.mosip.kernel.auth.dto.UserOtp;
+import io.mosip.kernel.auth.dto.UserPasswordRequestDto;
+import io.mosip.kernel.auth.dto.UserPasswordResponseDto;
+import io.mosip.kernel.auth.dto.UserRegistrationRequestDto;
+import io.mosip.kernel.auth.dto.UserRoleDto;
+import io.mosip.kernel.auth.dto.ValidationResponseDto;
+import io.mosip.kernel.auth.dto.otp.OtpUser;
+import io.mosip.kernel.auth.exception.AuthManagerException;
+import io.mosip.kernel.auth.exception.LoginException;
+import io.mosip.kernel.auth.repository.UserStoreFactory;
+import io.mosip.kernel.auth.repository.impl.KeycloakImpl;
+import io.mosip.kernel.auth.service.AuthService;
+import io.mosip.kernel.auth.service.OTPService;
+import io.mosip.kernel.auth.service.TokenService;
+import io.mosip.kernel.auth.service.UinService;
+import io.mosip.kernel.auth.util.ProxyTokenGenerator;
+import io.mosip.kernel.auth.util.TokenGenerator;
+import io.mosip.kernel.auth.util.TokenValidator;
+import io.mosip.kernel.core.util.EmptyCheckUtils;
+
+/**
+ * Auth Service for Authentication and Authorization
+ * 
+ * @author Ramadurai Pandian
+ * @author Urvil Joshi
+ * @author Srinivasan
+ *
```

Comment:
Can you please add a clear warning in the comment that this is only for local profile and should never be used in production. this is for developers to help test their environment.

Reply:
@Profile('local') annotation is there which already a marker for this statement still I will add.